Databricks on Azure with PrivateLink - Simplified deployment (#1977)
databricks/terraform-provider-databricks: docs/guides/azure-private-link-workspace-simplified.md

---

> page_title: "Provisioning Databricks on Azure with PrivateLink - Simplified deployment"

Reviewer: if you run `make fmt-docs`, it will format all the Terraform snippets in the doc

> page_title: "Provisioning Databricks on Azure with PrivateLink - Simplified deployment"

Reviewer: ```suggestion
page_title: "Provisioning Databricks on Azure with Private Link - Simplified deployment"
```

It is Private Link (two words), let's just make sure it is consistent across this guide

---

# Deploying pre-requisite resources and enabling PrivateLink connections

> -> **Note** This guide assumes that connectivity from the on-premises user environment is already configured using Expressroute or a VPN gateway connection.

Reviewer: ```suggestion
-> **Note** This guide assumes that connectivity from the on-premises user environment is already configured using ExpressRoute or a VPN gateway connection.
```

Databricks PrivateLink support enables private connectivity between users and their Databricks workspaces and between clusters on the data plane and core services on the control plane within the Databricks workspace infrastructure. 

You can use Terraform to deploy the underlying cloud resources and the private access settings resources automatically, using a programmatic approach. 

> This guide covers a [simplified deployment](https://learn.microsoft.com/en-us/azure/databricks/administration-guide/cloud-configurations/azure/private-link-simplified) to configure Azure Databricks with private link:

Reviewer: ```suggestion
This guide covers a [simplified deployment](https://learn.microsoft.com/en-us/azure/databricks/administration-guide/cloud-configurations/azure/private-link-simplified) to configure Azure Databricks with Private link:
```

* No separate VNet separates user access from the VNet that you use for your compute resources in the Classic data plane
* A transit subnet in the data plane VNet is used for user access
* Only a single private endpoint is used for both front-end and back-end connectivity. 

> * A seperate private endpoint is used for web authentication

Reviewer: ```suggestion
* A separate private endpoint is used for web authentication
```

* The same Databricks workspace is used for web authentication traffic but Databricks strongly recommends creating a separate workspace called a private web auth workspace for each region to host the web auth private network settings.

> ![Private Link backend](../images/azure-private-link-simplified.png)

Reviewer 2: ```suggestion
![Private Link backend](https://github.com/databricks/terraform-provider-databricks/raw/master/docs/images/azure-private-link-simplified.png)
```

This guide uses the following variables:

- `cidr`: The CIDR for the Azure Vnet
- `rg_name`: The name of the existing resource group
- `location`: The location for Azure resources

This guide is provided as-is and you can use it as the basis for your custom Terraform module.

To get started with Azure PrivateLink integration, this guide takes you through the following high-level steps:

- Initialize the required providers
- Configure Azure objects
  - Deploy an Azure Vnet with the following subnets:
    - Public and private subnets for Azure Databricks workspace
    - Private link subnet that will contain the following private endpoints:
      - Frontend / Backend private endpoint
      - Web_auth private endpoint 

>   - Configure the private dns zone in order to add: 

Reviewer: ```suggestion
  - Configure the private DNS zone in order to add: 
```

    - DNS A record to map connection for workspace access
    - DNS A record(s) for web_auth
- Workspace Creation

## Provider initialization

Initialize provider 

```hcl
terraform {
  required_providers {
    databricks = {
      source = "databricks/databricks"
    }
    azurerm = {
      source  = "hashicorp/azurerm"

>       version = ">=2.83.0"

Reviewer: can we pin azurerm higher? it is currently at 3.x already

    }
  }
}

provider "azurerm" {
  features {}
}
```

Define the required variables

```hcl
variable "cidr" {
  type    = string
}

variable "rg_name" {
  type    = string
}

variable "location" {
  type    = string
}

data "azurerm_client_config" "current" {
}

data "external" "me" {
  program = ["az", "account", "show", "--query", "user"]
}

locals {
  prefix = "abd-pl"
  tags = {
    Environment = "Demo"
    Owner       = lookup(data.external.me.result, "name")
  }
}
```

## Configure network 

### Deploy Azure vnet and subnets

> Create a new Azure vnet and the different required subnets :

Reviewer: ```suggestion
Create a new Azure VNet, the required subnets and associated security groups:
```
